src/main/java/org/jabref/logic/exporter/ModsExporter.java

```diff
@@ -76,267 +49,346 @@ public void export(final BibDatabaseContext databaseContext, final Path file, Li
         }
 
         try {
-            ModsCollectionDefinition modsCollection = new ModsCollectionDefinition();
+            // create writer
+            IndentingXMLStreamWriter writer = createWriter(file);
+
             for (BibEntry bibEntry : entries) {
-                ModsDefinition mods = new ModsDefinition();
-                bibEntry.getCitationKey().ifPresent(citeKey -> addIdentifier(new UnknownField("citekey"), citeKey, mods));
+                bibEntry.getCitationKey().ifPresent(citeKey -> {
+                    try {
+                        addIdentifier(writer, new UnknownField("citekey"), citeKey);
+                    } catch (
+                            XMLStreamException e) {
+                        throw new RuntimeException(e); // cant throw SaveException?
+                    }
+                });
+                if (!bibEntry.getCitationKey().isPresent()) {
```

Comment:
use isEmpty instead

Reply:
Can be merged with if statement, can't it?